src/com/google/javascript/jscomp/PeepholeRemoveDeadCode.java

```diff
@@ -296,6 +296,55 @@ public boolean apply(Node n) {
 
   static final Predicate<Node> MATCH_UNNAMED_BREAK = new MatchUnnamedBreak();
 
+  private Node tryRemoveSwitchWithSingleCase(Node n, boolean shouldIncludeCondition) {
+    Node caseBlock = n.getLastChild().getLastChild();
+    // Drop statements that are dead code
+    boolean shouldDrop = false;
+    for (Node statement: caseBlock.children()) {
```

Comment:
Since this particular code pattern seems pretty unrealistic for people to write (obviously dead code in the middle of a case), I'd probably skip this optimization it if it simplifies our implementation code.